File: AtCoder/abc201/C.swift
let s = readLine() ?? ""
let S = Array(s)
var answer = 0
for i in 0...9999 {
	var curr = i.toString()
	var good = true
	for d in 0...9 {
		if S[d] == "o" {
			var count = 0
			curr.forEach {
				if (($0.wholeNumberValue ?? 0) ?? 0) == d {
					count += 1
				}
			}
			if count == 0 {
				good = false
			}
		} else if S[d] == "x" {
			curr.forEach {
				if (($0.wholeNumberValue ?? 0) ?? 0) == d {
					good = false
				}
			}
		}
 	}
	if (good) {
		answer += 1
	}
}
print(answer)

extension Int {
	func toString() -> String {
		var ans = String(self)
		while ans.count < 4 {
			ans = "0" + ans
		}
		return ans
	}
}

File: AtCoder/abc201/C.kt
fun main() {
    val s = readLine()!!
    var ans = 0
    for (i in 0 until 10000) {
        var curr = i.toString()
        while (curr.length < 4) curr = "0$curr"
        var good = true
        for (d in 0 until 10) {
            if (s[d] == 'o') {
                var cnt = 0
                curr.forEach {
                    if (it - '0' == d) ++cnt
                }
                if (cnt == 0) good = false
            } else if (s[d] == 'x') {
                curr.forEach {
                    if (it - '0' == d) good = false
                }
            }
        }
        if (good) ++ans
    }
    print(ans)
}
